### PR TITLE
Adjust code to support newer bullmq

### PR DESCRIPTION
**Pull request by kibertoad**
## Changes

There were some changes around generics in newer BullMQ, this makes our code compatible with it.

## Checklist

- [X] Apply one of following labels; `major`, `minor`, `patch` or `skip-release`
- [X] I've updated the documentation, or no changes were necessary
- [X] I've updated the tests, or no changes were necessary

### CODE REVIEW

File: packages/app/background-jobs-common/src/background-job-processor/types.ts [hunk 1]
```diff
@@ -20,28 +20,6 @@ export type SafeJob<T = any, R = any, N extends string = string> = Omit<
   waitUntilFinished(queueEvents: any, ttl?: number): Promise<R>
 }
 
-export type SafeQueue<
```
Comment (kibertoad) [inline packages/app/background-jobs-common/src/background-job-processor/types.ts:23]: this is no longer necessary, BullMQ changes finally make types between pro and free versions more compatible out-of-the-box

File: packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts [hunk 2]
```diff
@@ -264,8 +264,10 @@ export abstract class AbstractBackgroundJobProcessor<
     const jobs =
       (await this._queue?.addBulk(
         jobData.map((data) => ({
-          name: this.config.queueId,
-          data,
+          // biome-ignore lint/suspicious/noExplicitAny: bull expects ExtractNameType<JobPayload>, but ExtractNameType is not exposed
+          name: this.config.queueId as any,
```
Comment (kibertoad) [inline packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts:268]: I hope there is a better solution for this, I would be grateful if someone could look deeper into how we can make TS happy here

**Review by CarlosGamero — APPROVED** [review]

File: packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts (hunk 2 above)
Comment (CarlosGamero) [reply packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts:268]: I will try to have a look later 🙏 But if you can't find a better solution I doubt I will be able to do it 😅

Comment (kibertoad) [reply packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts:268]: You are much better with TypeScript type wizardry than I am, so there is definitely hope :D

**Pull request merged**